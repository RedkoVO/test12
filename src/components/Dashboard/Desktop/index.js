import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import Transaction from '../components/Transaction'
import BestAd from '../components/BestAd'
import Categories from '../components/Categories'
import Wallet from '../components/Wallet'

import styles from './styles'

const Dashboard = ({
  classes,
  transactions,
  bestAds,
  gameCategories,
  allBalance,
  isDisabledButton,
  onSubmit,
  addressKey,
  handleChangeBalance,
  curencySelectValue
}) => (
  <main>
    <div className={classes.root}>
      <Wallet
        onSubmit={onSubmit}
        addressKey={addressKey}
        isDisabledButton={isDisabledButton}
        allBalanceResult={allBalance && allBalance.result}
        handleChangeBalance={handleChangeBalance}
        curencySelectValue={curencySelectValue}
      />
      <Transaction transactions={transactions} />
      <BestAd bestAds={bestAds} />
      <Categories categories={gameCategories} />
    </div>
  </main>
)

Dashboard.propTypes = {
  classes: PropTypes.object,
  transactions: PropTypes.array,
  bestAds: PropTypes.array,
  gameCategories: PropTypes.array,
  allBalance: PropTypes.object,
  isDisabledButton: PropTypes.bool,
  onSubmit: PropTypes.func,
  addressKey: PropTypes.string,
  handleChangeBalance: PropTypes.func,
  curencySelectValue: PropTypes.string
}

export default withStyles(styles)(Dashboard)
