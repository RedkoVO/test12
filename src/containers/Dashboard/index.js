import compose from 'recompose/compose'

import AsyncDashboardDesktop from '../../components/Dashboard/Desktop/AsyncDashboardDesktop'

export default compose()(AsyncDashboardDesktop)