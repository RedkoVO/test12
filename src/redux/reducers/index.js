import { combineReducers } from 'redux'
import { reducer as form } from 'redux-form'
import auth from './auth'
import balance from './balance'

export default combineReducers({
  form,
  auth,
  balance
})