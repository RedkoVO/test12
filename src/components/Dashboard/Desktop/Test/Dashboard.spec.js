import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { reduxForm } from 'redux-form'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

import Dashboard from '../'

const dataTransaction = [
  {
    id: 1,
    address: '0x88d50B466BfE55222019D71F9E8fAe17f5f45FCA1',
    amount: '0.1221 DCB',
    status: 'DEMO',
    time: '01:30 PM 12.02.2018'
  },
  {
    id: 2,
    address: '0x88d50B466BfE55222019D71F9E8fAe17f5f45FCA1',
    amount: '0.1221 DCB',
    status: 'DEMO',
    time: '01:30 PM 12.02.2018'
  },
  {
    id: 3,
    address: '0x88d50B466BfE55222019D71F9E8fAe17f5f45FCA1',
    amount: '0.1221 DCB',
    status: 'DEMO',
    time: '01:30 PM 12.02.2018'
  },
  {
    id: 4,
    address: '0x88d50B466BfE55222019D71F9E8fAe17f5f45FCA1',
    amount: '0.1221 DCB',
    status: 'DEMO',
    time: '01:30 PM 12.02.2018'
  },
  {
    id: 5,
    address: '0x88d50B466BfE55222019D71F9E8fAe17f5f45FCA1',
    amount: '0.1221 DCB',
    status: 'DEMO',
    time: '01:30 PM 12.02.2018'
  },
  {
    id: 6,
    address: '0x88d50B466BfE55222019D71F9E8fAe17f5f45FCA1',
    amount: '0.1221 DCB',
    status: 'DEMO',
    time: '01:30 PM 12.02.2018'
  }
]
const dataBestAds = [
  {
    id: 1,
    title: 'Games',
    img: 'CloverLogo1',
    link: '/games',
    titleLink: 'Game of the day',
    nameLink: 'Smash Champs'
  },
  {
    id: 2,
    title: 'Shop',
    img: 'CloverLogo1',
    link: '/stream',
    titleLink: 'Product of the day',
    nameLink: 'Blue cool Jelly'
  }
]
const dataCategories = [
  {
    id: 1,
    game: [
      {
        id: 1,
        title: 'Most popular',
        name: '4Clover',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 2,
        title: 'Most popular',
        name: 'Dice games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 3,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 4,
        title: 'For children',
        name: 'Funny games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 5,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 6,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      }
    ]
  },
  {
    id: 2,
    game: [
      {
        id: 1,
        title: 'Most popular',
        name: 'Loto games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 2,
        title: 'Most popular',
        name: 'Dice games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 3,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      }
    ]
  }
]

const Wrapper = reduxForm({ form: 'test' })(({ children }) => children)
const store = createStore(a => a, {})

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Wrapper>
            <Dashboard
              transactions={dataTransaction}
              bestAds={dataBestAds}
              gameCategories={dataCategories}
              onSubmit={() => {}}
              isDisabledButton={false}
              balance={{}}
            />
          </Wrapper>
        </MemoryRouter>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
