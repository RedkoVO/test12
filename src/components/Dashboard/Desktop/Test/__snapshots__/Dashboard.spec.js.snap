// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <div
    className="Dashboard-root-1"
  >
    <div
      className="Wallet-root-2"
    >
      <div
        className="Wallet-wallet-3"
      >
        <div
          className="Wallet-walletTitle-4"
        >
          <p>
            My Address:
          </p>
        </div>
        <div
          className="Wallet-walletAddress-5"
        />
        <div
          className="MuiInputBase-root-33 MuiInput-root-20 MuiInput-underline-24 Wallet-selectCurency-6"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root-13"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="Wallet-selectField-8"
              id="select-curency"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              id="curencyDropdownWallet"
              name="curency"
              type="hidden"
              value="DCB"
            />
            <i
              className="Wallet-curencyIcon-9"
            />
          </div>
        </div>
        <ul
          className="Wallet-walletNav-11"
        >
          <li>
            <div
              className="Wallet-walletNavItem-12"
              onClick={[Function]}
            >
              DCB
            </div>
          </li>
          <li>
            <div
              className="Wallet-walletNavItem-12"
              onClick={[Function]}
            >
              USD
            </div>
          </li>
          <li>
            <div
              className="Wallet-walletNavItem-12"
              onClick={[Function]}
            >
              EUR
            </div>
          </li>
          <li>
            <div
              className="Wallet-walletNavItem-12"
              onClick={[Function]}
            >
              CPT
            </div>
          </li>
        </ul>
      </div>
      <div
        className="Action-root-54"
      >
        <div
          className="Action-actionsSend-59"
        >
          <form
            className="Action-form-57"
            onSubmit={[Function]}
          >
            <span
              className="Action-sendTitle-60"
            >
              Send now
            </span>
            <div
              className="Action-wrSendField-61"
            >
              <label
                htmlFor="addressKey"
              >
                Address:
              </label>
              <input
                className="Action-sendAddressField-62"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Put address you want to send the money"
                type="text"
                value=""
              />
            </div>
            <div
              className="Action-wrSendField-61"
            >
              <label
                htmlFor="amount"
              >
                Amount:
              </label>
              <input
                className="Action-sendAmmountField-63"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Set ammount"
                type="text"
                value=""
              />
            </div>
            <button
              className="Action-sendBtn-65"
              type="submit"
            >
              Send money
            </button>
          </form>
        </div>
        <div
          className="Action-actionsInvoice-67"
        >
          <span
            className="Action-invoiceTitle-68"
          >
            Invoice
          </span>
          <div
            className="Action-wrSendField-61"
          >
            <label
              htmlFor="invoice"
            >
              Email:
            </label>
            <input
              className="Action-invoiceAmmountField-64"
              name="invoice"
              placeholder="Set ammount you need to receive"
              type="text"
            />
          </div>
          <button
            className="Action-sendBtn-65"
            type="submit"
          >
            Send Invoice
          </button>
        </div>
      </div>
    </div>
    <div
      className="BestAd-root-69"
    >
      <div
        className="BestAd-bestItem-70"
      >
        <div
          className="BestAd-titleWrap-71"
        >
          <a
            className="BestAd-bestTitle-72"
            href="/games"
            onClick={[Function]}
          >
            Games
          </a>
        </div>
        <div
          className="BestAd-bestContent-73"
        >
          <a
            className="BestAd-bestImg-74"
            href="/games"
            onClick={[Function]}
          >
            <img
              alt=""
              className="BestAd-bestItemImg-75"
              src="CloverLogo1"
            />
          </a>
          <a
            className="BestAd-bestLink-76"
            href="/games"
            onClick={[Function]}
          >
            Game of the day
            : 
            <br />
            <span>
              Smash Champs
            </span>
          </a>
        </div>
      </div>
      <div
        className="BestAd-bestItem-70"
      >
        <div
          className="BestAd-titleWrap-71"
        >
          <a
            className="BestAd-bestTitle-72"
            href="/stream"
            onClick={[Function]}
          >
            Shop
          </a>
        </div>
        <div
          className="BestAd-bestContent-73"
        >
          <a
            className="BestAd-bestImg-74"
            href="/stream"
            onClick={[Function]}
          >
            <img
              alt=""
              className="BestAd-bestItemImg-75"
              src="CloverLogo1"
            />
          </a>
          <a
            className="BestAd-bestLink-76"
            href="/stream"
            onClick={[Function]}
          >
            Product of the day
            : 
            <br />
            <span>
              Blue cool Jelly
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      className="Categories-root-77"
    >
      <div
        className="Categories-categories-78"
      >
        <div
          className="Categories-categTitleWrap-79"
        >
          <a
            className="Categories-categoriesTitle-80"
            href="/"
            onClick={[Function]}
          >
            Categories:
          </a>
        </div>
        <div
          className="CategorieItem-root-81"
        >
          <a
            className="CategorieItem-gameImg-82"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-83"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-84"
            href="/game/4clover"
            onClick={[Function]}
          >
            Most popular
            : 
            <br />
             
            <span>
              4Clover
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-81"
        >
          <a
            className="CategorieItem-gameImg-82"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-83"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-84"
            href="/game/4clover"
            onClick={[Function]}
          >
            Most popular
            : 
            <br />
             
            <span>
              Dice games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-81"
        >
          <a
            className="CategorieItem-gameImg-82"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-83"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-84"
            href="/game/4clover"
            onClick={[Function]}
          >
            For children
            : 
            <br />
             
            <span>
              Word games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-81"
        >
          <a
            className="CategorieItem-gameImg-82"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-83"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-84"
            href="/game/4clover"
            onClick={[Function]}
          >
            For children
            : 
            <br />
             
            <span>
              Funny games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-81"
        >
          <a
            className="CategorieItem-gameImg-82"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-83"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-84"
            href="/game/4clover"
            onClick={[Function]}
          >
            For children
            : 
            <br />
             
            <span>
              Word games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-81"
        >
          <a
            className="CategorieItem-gameImg-82"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-83"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-84"
            href="/game/4clover"
            onClick={[Function]}
          >
            For children
            : 
            <br />
             
            <span>
              Word games
            </span>
          </a>
        </div>
      </div>
      <div
        className="Categories-categories-78"
      >
        <div
          className="Categories-categTitleWrap-79"
        >
          <a
            className="Categories-categoriesTitle-80"
            href="/"
            onClick={[Function]}
          >
            Categories:
          </a>
        </div>
        <div
          className="CategorieItem-root-81"
        >
          <a
            className="CategorieItem-gameImg-82"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-83"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-84"
            href="/game/4clover"
            onClick={[Function]}
          >
            Most popular
            : 
            <br />
             
            <span>
              Loto games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-81"
        >
          <a
            className="CategorieItem-gameImg-82"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-83"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-84"
            href="/game/4clover"
            onClick={[Function]}
          >
            Most popular
            : 
            <br />
             
            <span>
              Dice games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-81"
        >
          <a
            className="CategorieItem-gameImg-82"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-83"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-84"
            href="/game/4clover"
            onClick={[Function]}
          >
            For children
            : 
            <br />
             
            <span>
              Word games
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</main>
`;
